UserSyncTest does not clean LDAP properly

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

testsuite

### Describe the bug

This class:
https://github.com/keycloak/keycloak/blob/main/testsuite/model/src/test/java/org/keycloak/testsuite/model/user/UserSyncTest.java

Is currently failing when running both test with the following exception:
```
2023-08-22T01:03:13.5822812Z org.keycloak.models.ModelException: org.keycloak.models.ModelException: Error creating subcontext [uid=user1,ou=People,dc=keycloak,dc=org]
2023-08-22T01:03:13.5823977Z 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2023-08-22T01:03:13.5824656Z 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
2023-08-22T01:03:13.5825368Z 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2023-08-22T01:03:13.5825952Z 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
2023-08-22T01:03:13.5826423Z 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
2023-08-22T01:03:13.5826903Z 	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:562)
2023-08-22T01:03:13.5827409Z 	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:591)
2023-08-22T01:03:13.5827863Z 	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:689)
2023-08-22T01:03:13.5828304Z 	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
2023-08-22T01:03:13.5829177Z 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfInt.evaluateParallel(ForEachOps.java:188)
2023-08-22T01:03:13.5829899Z 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
2023-08-22T01:03:13.5830344Z 	at java.base/java.util.stream.IntPipeline.forEach(IntPipeline.java:463)
2023-08-22T01:03:13.5830767Z 	at java.base/java.util.stream.IntPipeline$Head.forEach(IntPipeline.java:620)
2023-08-22T01:03:13.5831350Z 	at org.keycloak.testsuite.model.user.UserSyncTest.testManyUsersImport(UserSyncTest.java:113)
2023-08-22T01:03:13.5831874Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-08-22T01:03:13.5832397Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2023-08-22T01:03:13.5832979Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-08-22T01:03:13.5833470Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2023-08-22T01:03:13.5833890Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2023-08-22T01:03:13.5834390Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2023-08-22T01:03:13.5834905Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2023-08-22T01:03:13.5835656Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2023-08-22T01:03:13.5836161Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2023-08-22T01:03:13.5836737Z 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
2023-08-22T01:03:13.5930706Z 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
2023-08-22T01:03:13.5931249Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2023-08-22T01:03:13.5931994Z 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
2023-08-22T01:03:13.5932786Z 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
2023-08-22T01:03:13.5933431Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
2023-08-22T01:03:13.5935003Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
2023-08-22T01:03:13.5935490Z 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
2023-08-22T01:03:13.5935877Z 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
2023-08-22T01:03:13.5936285Z 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
2023-08-22T01:03:13.5936694Z 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
2023-08-22T01:03:13.5937163Z 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
2023-08-22T01:03:13.5937624Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2023-08-22T01:03:13.5938084Z 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
2023-08-22T01:03:13.5938468Z 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
2023-08-22T01:03:13.5938847Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2023-08-22T01:03:13.5939228Z 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
2023-08-22T01:03:13.5939673Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:377)
2023-08-22T01:03:13.5940211Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:284)
2023-08-22T01:03:13.5940763Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:248)
2023-08-22T01:03:13.5941274Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:167)
2023-08-22T01:03:13.5941794Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
2023-08-22T01:03:13.5942315Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
2023-08-22T01:03:13.5942798Z 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
2023-08-22T01:03:13.5943254Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
2023-08-22T01:03:13.5946263Z Caused by: org.keycloak.models.ModelException: Error creating subcontext [uid=user1,ou=People,dc=keycloak,dc=org]
2023-08-22T01:03:13.5946886Z 	at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.createSubContext(LDAPOperationManager.java:654)
2023-08-22T01:03:13.5947865Z 	at org.keycloak.storage.ldap.idm.store.ldap.LDAPIdentityStore.add(LDAPIdentityStore.java:102)
2023-08-22T01:03:13.5948367Z 	at org.keycloak.storage.ldap.LDAPUtils.lambda$addUserToLDAP$1(LDAPUtils.java:114)
2023-08-22T01:03:13.5949187Z 	at org.keycloak.storage.ldap.idm.model.LDAPObject.executeConsumerOnMandatoryAttributesComplete(LDAPObject.java:259)
2023-08-22T01:03:13.5949907Z 	at org.keycloak.storage.ldap.idm.model.LDAPObject.executeOnMandatoryAttributesComplete(LDAPObject.java:81)
2023-08-22T01:03:13.5950467Z 	at org.keycloak.storage.ldap.LDAPUtils.addUserToLDAP(LDAPUtils.java:112)
2023-08-22T01:03:13.5950927Z 	at org.keycloak.storage.ldap.LDAPUtils.addUserToLDAP(LDAPUtils.java:75)
2023-08-22T01:03:13.5951401Z 	at org.keycloak.testsuite.util.LDAPTestUtils.addLDAPUser(LDAPTestUtils.java:147)
2023-08-22T01:03:13.5951925Z 	at org.keycloak.testsuite.model.user.UserSyncTest.lambda$testManyUsersImport$3(UserSyncTest.java:118)
2023-08-22T01:03:13.5952486Z 	at org.keycloak.testsuite.model.KeycloakModelTest.lambda$inComittedTransaction$14(KeycloakModelTest.java:606)
2023-08-22T01:03:13.5953097Z 	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithResult(KeycloakModelUtils.java:383)
2023-08-22T01:03:13.5953736Z 	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithResult(KeycloakModelUtils.java:366)
2023-08-22T01:03:13.5954344Z 	at org.keycloak.testsuite.model.KeycloakModelTest.inComittedTransaction(KeycloakModelTest.java:594)
2023-08-22T01:03:13.5954936Z 	at org.keycloak.testsuite.model.KeycloakModelTest.inComittedTransaction(KeycloakModelTest.java:566)
2023-08-22T01:03:13.5955498Z 	at org.keycloak.testsuite.model.user.UserSyncTest.lambda$testManyUsersImport$4(UserSyncTest.java:113)
2023-08-22T01:03:13.5955989Z 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfInt.accept(ForEachOps.java:204)
2023-08-22T01:03:13.5956580Z 	at java.base/java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(Streams.java:104)
2023-08-22T01:03:13.5957037Z 	at java.base/java.util.Spliterator$OfInt.forEachRemaining(Spliterator.java:711)
2023-08-22T01:03:13.5957484Z 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
2023-08-22T01:03:13.5957939Z 	at java.base/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
2023-08-22T01:03:13.5958436Z 	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
2023-08-22T01:03:13.5958879Z 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
2023-08-22T01:03:13.5959334Z 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
2023-08-22T01:03:13.5959791Z 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
2023-08-22T01:03:13.5960234Z 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
2023-08-22T01:03:13.5960707Z 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
2023-08-22T01:03:13.5961523Z Caused by: javax.naming.NameAlreadyBoundException: [LDAP: error code 68 - ENTRY_ALREADY_EXISTS: failed for MessageType : ADD_REQUEST
2023-08-22T01:03:13.5961915Z Message ID : 5
2023-08-22T01:03:13.5962122Z     Add Request :
2023-08-22T01:03:13.5962318Z Entry
2023-08-22T01:03:13.5962559Z     dn: uid=user1,ou=People,dc=keycloak,dc=org
2023-08-22T01:03:13.5962837Z     objectclass: inetOrgPerson
2023-08-22T01:03:13.5963103Z     objectclass: organizationalPerson
2023-08-22T01:03:13.5963344Z     uid: user1
2023-08-22T01:03:13.5963557Z     mail: user1@email.org
2023-08-22T01:03:13.5963766Z     sn: User1LN
2023-08-22T01:03:13.5963969Z     cn: User1FN
2023-08-22T01:03:13.5964184Z     ManageDsaITImpl Control
2023-08-22T01:03:13.5964575Z         Type OID    : '2.16.840.1.113730.3.4.2'
2023-08-22T01:03:13.5964870Z         Criticality : 'false'
2023-08-22T01:03:13.5965104Z '
2023-08-22T01:03:13.5965592Z : ERR_250_ENTRY_ALREADY_EXISTS uid=user1,ou=People,dc=keycloak,dc=org already exists!]; remaining name 'uid=user1,ou=People,dc=keycloak,dc=org'
2023-08-22T01:03:13.5966146Z 	at java.naming/com.sun.jndi.ldap.LdapCtx.mapErrorCode(LdapCtx.java:3255)
2023-08-22T01:03:13.5966581Z 	at java.naming/com.sun.jndi.ldap.LdapCtx.processReturnCode(LdapCtx.java:3206)
2023-08-22T01:03:13.5967027Z 	at java.naming/com.sun.jndi.ldap.LdapCtx.processReturnCode(LdapCtx.java:2997)
2023-08-22T01:03:13.5967439Z 	at java.naming/com.sun.jndi.ldap.LdapCtx.c_createSubcontext(LdapCtx.java:841)
2023-08-22T01:03:13.5967924Z 	at java.naming/com.sun.jndi.toolkit.ctx.ComponentDirContext.p_createSubcontext(ComponentDirContext.java:341)
2023-08-22T01:03:13.5968539Z 	at java.naming/com.sun.jndi.toolkit.ctx.PartialCompositeDirContext.createSubcontext(PartialCompositeDirContext.java:268)
2023-08-22T01:03:13.5969164Z 	at java.naming/javax.naming.directory.InitialDirContext.createSubcontext(InitialDirContext.java:231)
2023-08-22T01:03:13.5969758Z 	at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager$7.execute(LDAPOperationManager.java:636)
2023-08-22T01:03:13.5970350Z 	at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager$7.execute(LDAPOperationManager.java:633)
2023-08-22T01:03:13.5970945Z 	at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.execute(LDAPOperationManager.java:729)
2023-08-22T01:03:13.5971558Z 	at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.execute(LDAPOperationManager.java:709)
2023-08-22T01:03:13.5972164Z 	at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.execute(LDAPOperationManager.java:704)
2023-08-22T01:03:13.5972811Z 	at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.createSubContext(LDAPOperationManager.java:633)
```

The cause seems to be the tests do not cleanup LDAP after test methods. This is causing users are already stored in LDAP and the following test fails with duplicate exception.

### Version

main

### Expected behavior

not failing test

### Actual behavior

failing test

### How to Reproduce?

Run UserSyncTest:
```
mvn test -Plegacy-jpa-federation+ldap+infinispan -Dtest=UserSyncTest
```

### Anything else?

_No response_

--- a/testsuite/model/src/test/java/org/keycloak/testsuite/model/user/UserSyncTest.java
+++ b/testsuite/model/src/test/java/org/keycloak/testsuite/model/user/UserSyncTest.java
@@ -100,6 +100,12 @@ public class UserSyncTest extends KeycloakModelTest {
 
     @Override
     public void cleanEnvironment(KeycloakSession s) {
+        final RealmModel realm = s.realms().getRealm(realmId);
+
+        ComponentModel ldapModel = LDAPTestUtils.getLdapProviderModel(realm);
+        LDAPStorageProvider ldapFedProvider = LDAPTestUtils.getLdapProvider(s, ldapModel);
+        LDAPTestUtils.removeAllLDAPUsers(ldapFedProvider, realm);
+
         s.realms().removeRealm(realmId);
     }
 

